crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,252 @@
+use std::sync::Arc;
+
+use axum::response::{IntoResponse, Response};
+
+use database::Repository;
+use didcomm::{AttachmentData, Message};
+use hyper::StatusCode;
+use mongodb::bson::doc;
+use serde_json::{json, Value};
+
+use crate::{
+    model::stateful::entity::{Connection, RoutedMessage},
+    web::{error::MediationError, AppState, AppStateRepository},
+};
+
+use super::error::RoutingError;
+
+/// Mediator receives forwarded messages, extract the next field in the message body, and the attachments in the message
+/// then stores the attachment with the next field as key for pickup
+pub async fn mediator_forward_process(
+    state: &AppState,
+    payload: Message,
+) -> Result<Message, Response> {
+  
+    let result = handler(state, payload).await.unwrap();
+    Ok(result)
+}
+
+async fn checks(
+    message: &Message,
+    connection_repository: &Arc<dyn Repository<Connection>>,
+) -> Result<String, Response> {
+    let next = message.body.get("next").and_then(Value::as_str);
+    match next {
+        Some(next) => next,
+        None => {
+            let response = (StatusCode::BAD_REQUEST, RoutingError::MalformedBody.json());
+            return Err(response.into_response());
+        }
+    };
+    // Check if the client's did in mediator's keylist
+    let _connection = match connection_repository
+        .find_one_by(doc! {"keylist": doc!{ "$elemMatch": { "$eq": &next}}})
+        .await
+        .map_err(|_| MediationError::RepostitoryError)
+        .unwrap()
```

Comment:
What is the necessity of mapping the error then?
```suggestion
        .map_err(|_| MediationError::RepostitoryError)?
```

Reply:
because i want it to panic with the mapped error, this is not something i want to catch, i don't know if you have noticed but i cannot propagate the error too (which again this is the type of error not to be handled )

Reply:
The client need to know that there were an error when trying to save its payload(s). You should not just panicked the code.
About the propagation you just need to map the error to a `Response` error. It’s easy to do.

Reply:
This is exactly why i want to panic the code because this error will occur only when the repository is not set which i believe should not be told to the client and if this error occours there is no need continuing as it will panics elsewhere if handled